### PR TITLE
feat: implement dialog

### LINKED ISSUE #4427: Implement anonymous consents details modal dialog
# General

When a user clicks on "view details" button on the sticky pop-up dialog (implemented in #3899), a modal dialog should be presented to the with anonymous consents.

See master ticket for more details: #3800

Create a PR against the master feature branch.

## Wire frames

Wire frames - https://98qkkl.axshare.com/#g=1&p=d_consent_form_details_default](https://98qkkl.axshare.com/#g=1&p=d_consent_form_details_default
Visuals - [desktop](https://app.zeplin.io/project/5a836f2897c4a02f0105b093/screen/5d69672faafe26b5b8cc4ba8), [mobile](https://app.zeplin.io/project/5a836f2897c4a02f0105b093/screen/5d69672d684fa927fed5ae66)

## Dialog

The dialog should have the following elements:

- "Allow all" button - clicking on this button, the user gives his consent to the all anonymous consents displayed in the dialog. After clicking on this button, the dialog should _remain open_, but the state of the consents should change to reflect the decision.
- "Reject all" button - clicking on this button, the user withdraws all anonymous consents. After clicking on this button, the dialog should _remain open_, but the state of the consents should change to reflect the decision.
- `X` or close button - clicking on this button, the dialog is closed. 
- Consent list - this list is an input of the dialog component. Each consent in the list has an accordion and clicking on it reveals the consent's description. When the accordion expands, the arrow changes. See [this](https://98qkkl.axshare.com/#g=1&p=d_consent_form_details_exposed). 
- In case there are multiple consents that don't fit in the dialog height, a scroll bar should appear
- A footer appears at the bottom. Re-use the same footer that appears on the homepage ("Copyright © 2019 SAP SE or an SAP affiliate company. All rights reserved.")
- keep in mind that this is a modal dialog, so clicking on the outside of the dialog should **not** close it.

## UI

In the current _Sparta_ theme, we're going to use check-boxes to indicate consents state. But when the new theme is applied, those check-boxes should be rendered as toggles.

For styling see [this ticket](https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/4398)

## Component structure

There should be two components implemented:

- `AnonymousConsentDetailsDialogComponent` - a container component that receives the consents as an `@Input()` and does the logic for giving / withdrawing a consent. At the same time it renders a list of `AnonymousConsentComponent` components, the close button, "Reject all" and "Allow all" button. At the bottom it should render the footer. In case the consent list is to big to fit the modal's size, a scroll bar appears.
- `AnonymousConsentComponent` - a presentational component that receives an individual consent via `@Input()` and renders it. The give / withdraw functionality is propagated to the `AnonymousConsentDetailsDialogComponent` component via `@Output()`. This component also renders the accordion that shows/hides the consent's description; this functionality is handled by this component. In case the consents description is too big to fit the modal's size, a scroll bar appears.

### PR DESCRIPTION
**Pull request by znikola**
closes #4427

### CODE REVIEW

File: projects/storefrontlib/src/shared/components/anonymous-consents/dialog/anonymous-consents-dialog.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      expect(
+        anonymousConsentsService.giveAnonymousConsent
+      ).toHaveBeenCalledTimes(mockTemplates.length);
+      expect(component.closeModal).toHaveBeenCalledWith('allowAll');
+      expect(component['subscriptions'].add).toHaveBeenCalled();
+    });
+  });
+
+  describe('onConsentChange', () => {
+    describe('whe the consent was given', () => {
```
Comment (LTiger14) [inline projects/storefrontlib/src/shared/components/anonymous-consents/dialog/anonymous-consents-dialog.component.spec.ts]: Typo in `when`

File: projects/storefrontlib/src/shared/components/anonymous-consents/dialog/form/anonymous-consent-form.component.scss [hunk 2]
```diff
@@ -0,0 +1,209 @@
+/* Checkboxes should work without toggle button styling
```
Comment (LTiger14) [inline projects/storefrontlib/src/shared/components/anonymous-consents/dialog/form/anonymous-consent-form.component.scss]: What is this?

File: projects/core/src/anonymous-consents/http-interceptors/anonymous-consents-interceptor.ts [hunk 3]
```diff
@@ -29,6 +31,10 @@ export class AnonymousConsentsInterceptor implements HttpInterceptor {
       take(1),
       withLatestFrom(this.authService.isUserLoggedIn()),
       switchMap(([consents, isUserLoggedIn]) => {
+        if (!this.isOccUrl(request.url)) {
+          return next.handle(request);
+        }
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/core/src/anonymous-consents/http-interceptors/anonymous-consents-interceptor.ts]: Prevents sending the handler to non-OCC endpoints, i.e. to the SOP mock endpoint.

File: projects/storefrontlib/src/shared/components/anonymous-consents/dialog/form/anonymous-consent-form.component.scss (hunk 2 above)
Comment (znikola) [reply projects/storefrontlib/src/shared/components/anonymous-consents/dialog/form/anonymous-consent-form.component.scss]: Miguel will take care of styles.

Comment (LTiger14) [reply projects/storefrontlib/src/shared/components/anonymous-consents/dialog/form/anonymous-consent-form.component.scss]: Ok but this file shouldn't be here. Hope he removes it 😂

File: projects/storefrontlib/src/shared/components/anonymous-consents/dialog/anonymous-consents-dialog.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    describe('when the consent was given', () => {
+      it('should call giveAnonymousConsent', () => {
+        spyOn(anonymousConsentsService, 'giveAnonymousConsent').and.stub();
+        component.onConsentChange({ given: true, template: mockTemplates[0] });
+        expect(
+          anonymousConsentsService.giveAnonymousConsent
+        ).toHaveBeenCalledWith(mockTemplates[0].id);
+      });
+    });
+    describe('whe the consent was withdrawn', () => {
```
Comment (LTiger14) [inline projects/storefrontlib/src/shared/components/anonymous-consents/dialog/anonymous-consents-dialog.component.spec.ts]: Typo in when

File: projects/storefrontlib/src/shared/components/anonymous-consents/dialog/form/anonymous-consent-form.component.scss (hunk 2 above)
Comment (znikola) [reply projects/storefrontlib/src/shared/components/anonymous-consents/dialog/form/anonymous-consent-form.component.scss]: 🤞

**Pull request merged**